### PR TITLE
Prettify ChunkedString and FileIO

### PR DESCRIPTION
**Pull request by rhazarian**
Encapsulation, memory leaks and useless actions fixed.

### CODE REVIEW

File: wurst/network/SyncSimple.wurst [hunk 1]
```diff
@@ -103,4 +104,3 @@ init
 				syncQueue.dequeue()
 				destroy syncData
 		checkQueue()
-
```
Comment (Frotty) [inline wurst/network/SyncSimple.wurst]: dont remove EOF newlines. editorconfig should enfore this if u use vscode.

File: wurst/file/SaveLoadData.wurst [hunk 2]
```diff
@@ -46,7 +46,7 @@ public interface LoadListener
 public function player.saveData(string slotName, string data)
 	let buffer = new ChunkedString(DEFAULT_CHUNK_SIZE)
 	buffer.append(data)
-	this.saveData(slotName.endsWith(".pld") ? slotName : slotName + ".pld", buffer)
```
Comment (Frotty) [inline wurst/file/SaveLoadData.wurst]: Last time i checked, `.pld` ending was required for the fileIO to work.
Have you tested this?

Comment (rhazarian) [reply wurst/file/SaveLoadData.wurst]: It is required, but it’s handled by the other overload of the function - the one taking a ChunkedString. The edited overloaded basically just converts a string into a ChunkedString and the file ending was handled twice.

Comment (rhazarian) [reply wurst/file/SaveLoadData.wurst]: It is required, but it’s handled by the other overload of the function - the one taking a ChunkedString. The edited overloaded basically just converts a string into a ChunkedString and the file ending was handled twice.

**Comment by Frotty** [comment]
Looks good, but I'm not a big fan of the breaking getter changes.

**Comment by rhazarian** [comment]
I can remove it if you want.
Though I think that it may be a good moment to change it now, because I guess none have used it a lot (if used at all) yet, and now the implementation is exposed and it blocks possible future changes

**Comment by Frotty** [comment]
Thanks 👍 
Make sure to update your other PR please.

**Pull request merged**